### ISSUE TITLE
how to set different parameters of leaf growth for growth and regrowth alfalfa separately

### ISSUE THREAD

**Issue by Submarine-666** [APSIMInitiative/APSIMClassic#2157]
Dear friends,

We found a good model performance on the alfalfa biomass in the first year, but an underestimation during the following years. We referred to the article https://www.researchgate.net/publication/263003628_Simulation_of_growth_and_development_of_diverse_legume_species_in_APSIM , where different parameters node_app_rate were given for growth and regrowth conditions. We suppose that parameter matters but found only one node_app_rate (varied with GDD) was given in the lucerne.xml file. We want to figure out how to set two different  node_app_rate for growth and regrowth conditions.

Cheers

**Comment by peter-devoil** [comment]
There is some code to do this sort of thing the existing lucerne implementation- see the class_action and class_change parameters. Unfortunately I can't find a test/demo of its use though.

The intent was that lucerne subscribes to an event (class_action), yet on a quick glance, I can't see how it re-reads the parameters. So you might want to create an experiment to be sure the method was fully implemented (a lot of stuff was written that was never finished)

**Comment by Submarine-666** [comment]
Thanks for your reply.
We found a paragraph in the latter part of lucerne.xml (please see the pic pasted below), does it mean during the regrowth period, relative parameters (e.g. node_app_rate, rue, transp_eff_cf) would be read from this part?
![image](https://user-images.githubusercontent.com/104030072/173314386-92df62f7-1bd8-4961-b2b9-ec9aeb45982c.png)

**Comment by peter-devoil** [comment]
Yes, that's the intent. Unfortunately, the change that introduced this code pre-dates version control, so I can't offer you the authors name, though you can do a with/without experiment to see if its working as expected..
